Adds additional product actions for Scan and Backup

ChaosExAnima opened the pull request:
#### Changes proposed in this Pull Request

* This adds the capability for additional buttons for Scan and Backup to link to the Cloud.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Navigate to Plan > My Plan page on a site with Scan or Backup purchased.
* Click relevant links.

#### Screenshots

Product Management
<img width="1043" alt="Screen Shot 2020-05-13 at 12 20 14 PM" src="https://user-images.githubusercontent.com/1760168/81838703-7d29d700-9514-11ea-826f-26e404e64d6b.png">

Plan Management
<img width="1074" alt="Screen Shot 2020-05-18 at 12 44 17 PM" src="https://user-images.githubusercontent.com/1760168/82238626-9a401a80-9905-11ea-98f3-1094dd5b6f68.png">

Fixes 1143508703416848-as-1164157450632305.

ChaosExAnima commented:
@jeffgolenski - FYI, I ended up using existing styles for the buttons which aren't quite like your mocks. Please let me know if you need any tweaks.

rcanepa commented:
Tests passed on my end. This makes me wonder if we should also display those buttons when a user has JP Premium which includes Backup and Scan (if I recall correctly). For instance, I have JP Premium and JP Scan, and I don't see the link to Backup (because I don't have that product).
![image](https://user-images.githubusercontent.com/3418513/81848442-81adba00-952b-11ea-8cd4-c5426783bf44.png)

ChaosExAnima commented:
@rcanepa - Great question. I'd also wonder about access to Cloud when not in the My Plan page, as that's a place I suspect the users won't go very often? More a design question, perhaps :)

enejb commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -202,6 +206,49 @@ class PurchasesListing extends Component {
+					) }
+					compact
+				>
+					{ translate( 'View Backups' ) }
```
Lets use sentence case.

ChaosExAnima replied:
Resolved in 5c63e51.

enejb commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -202,6 +206,49 @@ class PurchasesListing extends Component {
+					) }
+					compact
+				>
+					{ translate( 'Run Scan' ) }
```
Here as well, lets use sentence case.

ChaosExAnima replied:
Resolved in 5c63e51.

enejb commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -202,6 +206,49 @@ class PurchasesListing extends Component {
+			serviceButton = (
+				<Button
+					href={ addQueryArgs(
+						{ site, source: 'jetpack-scan-dash' },
```
this should be calypso-scanner instead of jetpack-scan-dash

ChaosExAnima replied:
Resolved in 3f770fa.

enejb commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -202,6 +206,49 @@ class PurchasesListing extends Component {
+			serviceButton = (
+				<Button
+					href={ addQueryArgs(
+						{ site, source: 'jetpack-backup-dash' },
```
I am not sure what this needs to be.

ChaosExAnima replied:
Talked with folks and I resolved in 3f770fa.

enejb commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -188,7 +192,7 @@ class PurchasesListing extends Component {
 
 		if ( isJetpackProduct( purchase ) ) {
-			label = translate( 'Manage product' );
+			label = translate( 'Manage Subscription' );
```
Lets use sentence case.

ChaosExAnima replied:
Resolved in 5c63e51.

a8ci18n commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -188,7 +192,7 @@ class PurchasesListing extends Component {
 
 		if ( isJetpackProduct( purchase ) ) {
-			label = translate( 'Manage product' );
+			label = translate( 'Manage subscription' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **32** times:
`translate( 'Manage Subscription' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/product-actions)

ChaosExAnima commented:
Updated links to not use redirects lib to rely on internal linking. Also updated copy to reflect users cannot run scans, just view and fix threats:
<img width="1043" alt="Screen Shot 2020-05-15 at 12 51 38 PM" src="https://user-images.githubusercontent.com/1760168/82075888-f9eeb980-96aa-11ea-83a8-68e7284748d1.png">

enejb commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -202,6 +204,33 @@ class PurchasesListing extends Component {
+		} else if ( isJetpackScan( purchase ) ) {
+			serviceButton = (
+				<Button href={ `/activity-log/${ site }?group=rewind` } compact>
+					{ translate( 'View threats' ) }
```
Since we don't know if a site has threats we shouldn't tell the user to go see them. How about View scan results?

ChaosExAnima replied:
@enejb - We don't show that, though? The only thing that shows in the activity log is a threat if one is present.

enejb commented on client/my-sites/plans/current-plan/purchases-listing.jsx:
```diff
@@ -202,6 +204,33 @@ class PurchasesListing extends Component {
+			);
+		} else if ( isJetpackScan( purchase ) ) {
+			serviceButton = (
+				<Button href={ `/activity-log/${ site }?group=rewind` } compact>
```
the `?group=rewind` is not needed here.

ChaosExAnima replied:
Revised in 826fc8b.

jeffgolenski commented:
@ChaosExAnima RE: button styles. Absolutely, no worries about matching the mocks perfectly. Using pre-existing components is the way to go! 

RE: plan buttons. @rcanepa Thanks for bringing this up. When a customer has a plan that includes backup, scan, or both, Instead of adding an entire section for Backup & Scan, can we just add a button to the right of "Manage Plan" that says "Manage Backup & Scan [external link icon]"?

ChaosExAnima commented:
Okay, I've updated things to show a button on the plan if a user has Backup and Scan. It updates if there's somehow a plan that has only one or the other, but that's futureproofing more than anything really.

I didn't include the external link like @jeffgolenski asked for as we aren't directing users to Cloud during the limited launch, so it'll remain local.

One potential issue is we're going to show this based on plan and not the Rewind API, so it's possible VaultPress users could be caught up in this. Thoughts?

jeffgolenski commented:
@ChaosExAnima I tested on a JN site with a pro plan, and also another JN site with just jetpack backup. Works wonderfully. Thank you!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3646217

Thank you @ChaosExAnima for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3646217).
